src/core/models/result.js

```diff
@@ -101,4 +104,28 @@ export default class Result {
      */
     this.distanceFromFilter = data.distanceFromFilter || null;
   }
+
+  /**
+   * Constructs an SDK Result from an answers-core Result
+   *
+   * @param {Result} result from answers-core
+   * @returns {@link Result}
+   */
+  static fromCore (result) {
+    const highlightedData = HighlightedFieldMap.fromCore(result.highlightedValues);
+    const details = highlightedData.description || result.description;
+    const truncatedDetails = truncate(details);
```

Comment:
Could truncating the highlighted data result in potentially invalid HTML if one of the `<strong>` tags is split?

Reply:
I can truncate before the highlighting is applied just in case

Reply:
After testing further, I learned that this works as-is because the truncate method splits on spaces, and the highlighted data seems to highlight each word rather than highlighted sequences of words which may contain spaces. The SDK currently truncates after applying highlighting, so it seems safe to do for now. The risk is if the backend every highlights a value which contains a space, it may break